email/config.go

```diff
@@ -2,11 +2,13 @@ package email
 
 type Config struct {
 	ValidateForm bool
+	ValidateDNS  bool
 }
 
 func newConfig() Config {
 	return Config{
```

Comment:
why not create a `type validate func(address string) (bool, error)` and then just add those to a map by name on the config. Then the Validator can just iterate the validate's on the configuration and create an error with the name of the validator on failure? That way you don't have to have a long sequence of `if` statements in the Validate